replica/server.go

```diff
@@ -159,25 +160,45 @@ func (s *Server) isExtentSupported() error {
 	return file.Close()
 }
 
+func (s *Server) initUUID() error {
+	var err error
+	r := &Replica{
+		dir:              s.Dir,
+		ReplicaStartTime: StartTime,
+		mode:             types.INIT,
+	}
+	if r.info, err = ReadInfo(s.Dir); err != nil {
+		return err
+	}
+	if r.info.UUID == "" {
+		UUID := guuid.New().String() + s.ReplicaAddress
+		hash := sha1.New()
+		hash.Write([]byte(UUID))
+		r.info.UUID = fmt.Sprintf("%x", hash.Sum(nil))
+		r.writeVolumeMetaData(r.info.Dirty, r.info.Rebuilding)
+	}
+	return nil
+}
+
 func (s *Server) Create(size int64) error {
 	s.Lock()
 	defer s.Unlock()
 	if err := s.isExtentSupported(); err != nil {
 		return err
 	}
+	defer s.initUUID()
```

Comment:
Why is this in `defer`? Shouldn;t it be executed prior to unlocking?

Reply:
Since the Unlock is also in differ, this will be executed before unlock (https://play.golang.org/p/-oUdvDmPhnT)

Reply:
cool! #TIL